x/inflation/types/params.go

```diff
@@ -7,96 +7,69 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	ethermint "github.com/tharsis/ethermint/types"
+	"gopkg.in/yaml.v2"
+
 	epochtypes "github.com/tharsis/evmos/x/epochs/types"
-	yaml "gopkg.in/yaml.v2"
 )
 
+// TODO refactor into iota
 // Parameter store keys
 var (
-	KeyMintDenom                            = []byte("MintDenom")
-	KeyGenesisEpochProvisions               = []byte("GenesisEpochProvisions")
-	KeyEpochIdentifier                      = []byte("EpochIdentifier")
-	KeyReductionPeriodInEpochs              = []byte("ReductionPeriodInEpochs")
-	KeyReductionFactor                      = []byte("ReductionFactor")
-	KeyPoolAllocationRatio                  = []byte("PoolAllocationRatio")
-	KeyDeveloperRewardsReceiver             = []byte("DeveloperRewardsReceiver")
-	KeyMintingRewardsDistributionStartEpoch = []byte("MintingRewardsDistributionStartEpoch")
+	KeyMintDenom              = []byte("KeyMintDenom")
+	KeyEpochIdentifier        = []byte("KeyEpochIdentifier")
+	KeyEpochsPerPeriod        = []byte("KeyEpochsPerPeriod")
+	KeyExponentialCalculation = []byte("KeyExponentialCalculation")
+	KeyInflationDistribution  = []byte("KeyInflationDistribution")
+	KeyTeamAddress            = []byte("KeyTeamAddress")
+	KeyTeamVestingProvision   = []byte("KeyTeamVestingProvision")
 )
 
-// ParamTable for minting module.
+// ParamTable for inflation module
 func ParamKeyTable() paramtypes.KeyTable {
 	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
 }
 
 func NewParams(
 	mintDenom string,
-	genesisEpochProvisions sdk.Dec,
 	epochIdentifier string,
-	ReductionFactor sdk.Dec,
-	reductionPeriodInEpochs int64,
-	distrProportions DistributionProportions,
-	weightedDevRewardsReceivers []WeightedAddress,
-	mintingRewardsDistributionStartEpoch int64,
+	epochsPerPeriod int64,
+	exponentialCalculation ExponentialCalculation,
+	inflationDistribution InflationDistribution,
+	teamAddress string,
```

Comment:
```suggestion
	teamAddress sdk.AccAddress,
```

Reply:
@fedekunze How do I declare a `sdk.AccAddress` in proto?
I tried to update the proto accordingly, but can't get it to work. I tried the following options
 
**Option 1**
```
  // address to receive the team allocation from the inflation distribution
  string team_address = 6 [
    (gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"
  ];
```

but getting the following error `/Users/danielburckhardt/code/tharsis/evmos/x/inflation/types/genesis.pb.go:171: cannot use "" (type string) as type "github.com/cosmos/cosmos-sdk/types".AccAddress in return argument`

**Option 2**
```
  // address to receive the team allocation from the inflation distribution
  string team_address = 6 [
    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"
    (gogoproto.nullable) = false
  ];
```

`/Users/danielburckhardt/code/tharsis/evmos/x/inflation/types/genesis.pb.go:278: m.TeamAddress.Size undefined (type *"github.com/cosmos/cosmos-sdk/types".AccAddress has no field or method Size)`